ignite/cmd/plugin.go

```diff
@@ -303,6 +314,20 @@ func linkPluginCmd(rootCmd *cobra.Command, p *plugin.Plugin, pluginCmd plugin.Co
 		// pluginCmd has no sub commands, so it's runnable
 		newCmd.RunE = func(cmd *cobra.Command, args []string) error {
 			return clictx.Do(cmd.Context(), func() error {
+				c, err := SetupChain(cmd)
+				if err != nil {
+					return err
+				}
+				manifest, err := p.Interface.Manifest()
+				if err != nil {
+					return err
+				}
+				if manifest.WithPaths {
+					if p.With == nil {
+						p.With = make(map[string]string)
+					}
+					p.With["appPath"] = c.AppPath()
```

Comment:
It might be good to explore if it's possible to maybe have `p.With` always initialized to avoid having to check an initialize it when new values are added. There might be more cases similar to this one in the future.

Reply:
> It might be good to explore if it's possible to maybe have `p.With` always initialized to avoid having to check and initialize it when new values are added. There might be more cases similar to this one in the future.

I assumed it was at first but got  a nil panic and realised that it isnt initialised if the config file does not contain additional data so that was a good fix. Ideally it should be initialized at the place where the Plugin struct is populated originally from the config file but couldnt find it